addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -55,12 +56,186 @@ def set_crm_team_message_types(env):
             (4, env.ref('sale.mt_salesteam_invoice_created').id)]})
 
 
+@openupgrade.logging()
+def set_so_line_amounts(env):
+    """We replicate here the code of the compute function and set the values
+    finally via SQL for avoiding the trigger of the rest of the computed
+    fields that depends on these fields.
+    """
+    lines = env['sale.order.line'].search([])
+    for line in lines:
+        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+        taxes = line.tax_id.compute_all(
+            price, line.order_id.currency_id, line.product_uom_qty,
+            product=line.product_id, partner=line.order_id.partner_id)
+        env.cr.execute("""
+            UPDATE sale_order_line
+            SET price_tax = %s,
+                price_total = %s,
+                price_subtotal = %s
+            WHERE id = %s""", (
+                taxes['total_included'] - taxes['total_excluded'],
+                taxes['total_included'],
+                taxes['total_excluded'],
+                line.id,
+            )
+        )
+
+
+def set_so_line_computed_rest(env):
+    """Emulate the computation of the rest of the computed fields through
+    equivalente SQL querys.
+    """
+    uom_precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line
+        SET price_reduce = CASE
+            -- ROUND() missing in original code, but it's a bug
+            WHEN ROUND(product_uom_qty, %(uom_precision)s) != 0
+                THEN price_subtotal / product_uom_qty
+            ELSE 0.0
+        END""",
+        {"uom_precision": uom_precision})
+    # Fail if there are UoM mismatches
+    openupgrade.logged_query(
+        env.cr,
+        """
+            SELECT
+                rel.order_line_id,
+                sol_u.id AS order_line_uom_id,
+                sol_u.category_id AS order_line_uom_category_id,
```

Comment:
I think you don't need to put the categories, and the SQL is vastly simplified. I would put maybe sales order number or product name though

Reply:
I prefer to keep the categories because it's actually gonna be the data that triggers this exception.

OTOH I'll add the other data.